### PR TITLE
Add Options page for disabling telemetry

### PR DESCRIPTION
**Pull request by RandomlyKnighted**
This is not yet ready to merge. It'll server as a placeholder while I finish up the last item. I've already added the options page. It's located under Tools->Options. 

All that's remaining is to read the saved setting from Visual Studio and send the data if the value is set to true. The issue I've run into is the Options page had to be added in a VSIX project, so I had to add it in the TemplatePack project. The code where I'll need to check the value and is in the GA code, which is in the LigerShark.Templates library. I'm not quite sure how to get the value since TemplatePack references LigerShark.Templates.

### CODE REVIEW

**Comment by RandomlyKnighted** [comment]
@sayedihashimi this is ready to go. I finished testing it this morning.

File: LigerShark.Templates/DynamicBuilder/SettingsStore.cs [hunk 1]
```diff
@@ -0,0 +1,34 @@
+﻿using Newtonsoft.Json;
+using System.IO;
+using System;
+
+namespace LigerShark.Templates.DynamicBuilder
+{
+    public class SettingsStore
+    {
+        [JsonProperty]
+        public bool SendTelemetry { get; set; }
```
Comment (sayedihashimi) [inline LigerShark.Templates/DynamicBuilder/SettingsStore.cs]: Can we default to true in the constructor?

File: LigerShark.Templates/DynamicBuilder/SettingsStore.cs [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+
+            if (!File.Exists(filePath))
+            {
+                throw new FileNotFoundException(string.Format(@"JSON settings file not found at [{0}]", filePath));
+            }
+
+            return JsonConvert.DeserializeObject<SettingsStore>(File.ReadAllText(filePath));
+        }
+
+        public void WriteJsonFile(string fileDirectory, string filePath, string json)
```
Comment (sayedihashimi) [inline LigerShark.Templates/DynamicBuilder/SettingsStore.cs]: Passing in the directory and file path separately may lead to bugs. Instead can you modify to accept filePath and then check the directory using the filePath. Just create a `FileInfo` object and call `path.Directory.Exists`.

Comment (RandomlyKnighted) [inline LigerShark.Templates/DynamicBuilder/SettingsStore.cs]: Sure I'll do that now.

**Comment by RandomlyKnighted** [comment]
@sayedihashimi Rather than defaulting SendTelemetry to true in the constructor I used the DefaultValue attribute so that it would default it to true automatically when the file is created.

**Pull request merged**